cloudwatchlogs/cloudwatchlogs_lambda.js

```diff
@@ -0,0 +1,215 @@
+        var metadataOverride = message._sumo_metadata;
+        if (metadataOverride.category) {
+            sourceCategory = metadataOverride.category;
+        }
+        if (metadataOverride.host) {
+            sourceHost = metadataOverride.host;
+        }
+        if (metadataOverride.source) {
+            sourceName = metadataOverride.source;
+        }
+        delete message._sumo_metadata;
+    }
+    return sourceName + ':' + sourceCategory + ':' + sourceHost;
+    
+}
+
+function postToSumo(context, messages) {
+    var messagesTotal = Object.keys(messages).length;
+    var messagesSent = 0;
+    var messageErrors = [];
+    
+    var urlObject = url.parse(SumoURL);
+    var options = {
+        'hostname': urlObject.hostname,
+        'path': urlObject.pathname,
+        'method': 'POST'
+    };
+    
+    var finalizeContext = function () {
+        var total = messagesSent + messageErrors.length;
+        if (total == messagesTotal) {
+            console.log('messagesSent: ' + messagesSent + ' messagesErrors: ' + messageErrors.length);
+            if (messageErrors.length > 0) {
+                context.fail('errors: ' + messageErrors);
+            } else {
+                context.succeed();
+            }
+        }
+    };
+    
+    
+    Object.keys(messages).forEach(function (key, index) {
+        var headerArray = key.split(':');
+        
+        options.headers = {
+            'X-Sumo-Name': headerArray[0],
+            'X-Sumo-Category': headerArray[1],
+            'X-Sumo-Host': headerArray[2]
+        };
+        
+        var req = https.request(options, function (res) {
+            var body = '';
+            res.setEncoding('utf8');
+            res.on('data', function (chunk) {
+                body += chunk;
+            });
+            res.on('end', function () {
+                if (res.statusCode == 200) {
+                    messagesSent++;
+                } else {
+                    errors.push('HTTP Return code ' + res.statusCode);
+                }
+                finalizeContext();
+            });
+        });
+        
+        req.on('error', function (e) {
+            messageErrors.push(e.message);
+            finalizeContext();
+        });
+        
+        for (var i = 0; i < messages[key].length; i++) {
+            req.write(JSON.stringify(messages[key][i]) + '\n');
+        }
+        req.end();
+    });
+}
+
+
+exports.handler = function (event, context) {
+    
+    // Validate URL has been set
+    var urlObject = url.parse(SumoURL);
+    if (urlObject.protocol != 'https:' || urlObject.host === null || urlObject.path === null) {
+        context.fail('Invalid SUMO_ENDPOINT environment variable: ' + SumoURL);
+    }
+    
+    var zippedInput = new Buffer(event.awslogs.data, 'base64');
+    
+    zlib.gunzip(zippedInput, function (e, buffer) {
+        if (e) {
+            context.fail(e);
+        }
+        
+        var awslogsData = JSON.parse(buffer.toString('ascii'));
```

Comment:
@kumoroku I agree with you, utf-8 seems to be far safer. All aws examples suggest ascii but reading http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/AgentReference.html suggests many encoding options. 

I've updated the code to support encoding as a param